test/cbmc/proofs/MQTTAgentCommand_Connect/MQTTAgentCommand_Connect_harness.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * coreMQTT-Agent v1.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file MQTTAgentCommand_Connect_harness.c
+ * @brief Implements the proof harness for MQTTAgentCommand_Connect function.
+ */
+
+/* MQTT agent include. */
+#include "mqtt_agent_command_functions.h"
+
+void harness()
+{
+    MQTTAgentContext_t * pMqttAgentContext;
+    MQTTAgentCommandFuncReturns_t * pReturnFlags;
+    MQTTAgentConnectArgs_t * pConnectArgs;
+
+    pMqttAgentContext = malloc( sizeof( MQTTAgentContext_t ) );
```

Comment:
Is there a reason why this struct is being allocated with arbitrary memory from the heap instead of using this function to allocate it?
https://github.com/FreeRTOS/coreMQTT-Agent/blob/main/test/cbmc/sources/mqtt_agent_cbmc_state.c#L66

Reply:
I have used it because in this function we are not using the message interface and some other members of the mqttAgentContext struct here in this function, so calling that function will be an unnnecessary overhead in terms of memory allocation, so didn't used it here.